digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java

```diff
@@ -295,7 +308,13 @@ protected TaskResult callExecutor(Path projectPath, String type, TaskRequest mer
         DefaultSecretProvider secretProvider = new DefaultSecretProvider(
                 secretContext, secretAccessPolicy, grants, operatorSecretFilter, secretStore);
 
-        TaskExecutionContext taskExecutionContext = new DefaultTaskExecutionContext(secretProvider);
+        PrivilegedVariables privilegedVariables = GrantedPrivilegedVariables.build(
+                mergedRequest.getLocalConfig().getNestedOrGetEmpty("_env"),
```

Comment:
How important is it restrict scope of this code path only to `sh>` operator? Why not `py>` and `rb>` scripts have chance to use secrets?